src/coreclr/jit/lsra.cpp

```diff
@@ -11726,8 +11773,6 @@ void LinearScan::verifyFreeRegisters(regMaskTP regsToFree)
             assert(nextIntervalRef[reg] == MaxLocation);
             assert(spillCost[reg] == 0);
         }
-        LsraLocation thisNextFixedRef = physRegRecord->getNextRefLocation();
```

Comment:
why is this not needed anymore?

Reply:
The assert isn't true anymore -- `nextFixedRef[reg]` has either the location of the next `RefTypeFixedReg` or of the next `RefTypeKill` that kills the register. But `physRegRecord->getNextRefLocation()` is always the next `RefTypeFixedReg` after this change.